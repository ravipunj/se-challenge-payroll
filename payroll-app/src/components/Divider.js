import React from 'react';

export const Divider = () => (
  <div className="row">
    <div className="column"><hr/></div>
  </div>
);