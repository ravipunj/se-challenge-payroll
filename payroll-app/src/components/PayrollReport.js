import React from 'react';
import axios from 'axios';

export class PayrollReport extends React.Component {
  constructor() {
    super();
    this.state = {payroll_report: []};
  }
  componentDidMount() {
    axios.get('/payroll_report').then(response => {
      if(response.status === 200) {
        this.setState({payroll_report: response.data})
      }
    });
  }
  render() {
    return (
      <div className="row">
        <div className="column">
          <table>
            <thead>
              <tr>
                <th>Employee ID</th>
                <th>Pay Period</th>
                <th>Amount Paid</th>
              </tr>
            </thead>
            <tbody>
              {this.state.payroll_report.map((entry,idx) => (
                <tr key={idx}>
                  <td>{entry.employee_id}</td>
                  <td>{entry.pay_period}</td>
                  <td>{entry.amount_paid}</td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}