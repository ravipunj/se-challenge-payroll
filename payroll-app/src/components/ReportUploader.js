import React from 'react';
import axios from 'axios';

export class ReportUploader extends React.Component {
  constructor() {
    super();
    this.state = {submitDisabled: true};
    this.onSubmit = this.onSubmit.bind(this);
    this.onFileChanged = this.onFileChanged.bind(this);
  }
  onSubmit(e) {
    e.preventDefault();
    const formData = new FormData();
    formData.append("report", this.fileInput.files[0]);
    axios.post("/payroll_csv", formData, {headers: {"Content-Type": "multipart/form-data"}});
  }
  onFileChanged() {
    this.setState({submitDisabled: this.fileInput.files.length === 0});
  }
  render() {
    return (
      <div className="row">
        <div className="column">
          <form onSubmit={this.onSubmit} style={style}>
            <fieldset>
              <h4>Upload Payroll Report CSV</h4>
              <input type="file" id="reportFile" ref={input => {this.fileInput = input}} onChange={this.onFileChanged}/>
              <input type="submit" className="button" value="Upload" disabled={this.state.submitDisabled}/>
            </fieldset>
          </form>
        </div>
      </div>
    );
  }
}

const style = {
  textAlign: 'center'
}