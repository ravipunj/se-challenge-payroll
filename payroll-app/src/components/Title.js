import React from 'react';

export const Title = () => (
  <div className="row">
    <div className="column" style={style}>
      <h1>Payroll</h1>
    </div>
  </div>
);

const style = {
  textAlign: 'center'
}