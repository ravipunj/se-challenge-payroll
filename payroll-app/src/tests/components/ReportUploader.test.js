import React from 'react';
import { mount } from 'enzyme';

import { ReportUploader } from '../../components/ReportUploader';

describe('ReportUploader', () => {
  it('matches snapshot', () => {
    expect(mount(<ReportUploader/>)).toMatchSnapshot();
  });
});