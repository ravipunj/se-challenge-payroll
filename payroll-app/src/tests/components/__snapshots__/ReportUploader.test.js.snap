// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportUploader matches snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="row"
>
        <div
                class="column"
        >
                <form
                        action="#"
                        style="text-align: center;"
                >
                        <fieldset>
                                <h4>
                                        Upload Payroll Report CSV
                                </h4>
                                <input
                                        type="file"
                                        id="reportFile"
                                />
                                <input
                                        type="submit"
                                        class="button-primary"
                                        value="Upload"
                                />
                        </fieldset>
                </form>
        </div>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="column"
>
          <form
                    action="#"
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
          >
                    <fieldset>
                              <h4>
                                        Upload Payroll Report CSV
                              </h4>
                              <input
                                        id="reportFile"
                                        type="file"
                              />
                              <input
                                        className="button-primary"
                                        type="submit"
                                        value="Upload"
                              />
                    </fieldset>
          </form>
</div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="column"
>
            <form
                        action="#"
                        style="text-align: center;"
            >
                        <fieldset>
                                    <h4>
                                                Upload Payroll Report CSV
                                    </h4>
                                    <input
                                                type="file"
                                                id="reportFile"
                                    />
                                    <input
                                                type="submit"
                                                class="button-primary"
                                                value="Upload"
                                    />
                        </fieldset>
            </form>
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              action="#"
              style={
                            Object {
                                          "textAlign": "center",
                                        }
              }
>
              <fieldset>
                            <h4>
                                          Upload Payroll Report CSV
                            </h4>
                            <input
                                          id="reportFile"
                                          type="file"
                            />
                            <input
                                          className="button-primary"
                                          type="submit"
                                          value="Upload"
                            />
              </fieldset>
</form>,
            "className": "column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <form
                action="#"
                style="text-align: center;"
>
                <fieldset>
                                <h4>
                                                Upload Payroll Report CSV
                                </h4>
                                <input
                                                type="file"
                                                id="reportFile"
                                />
                                <input
                                                type="submit"
                                                class="button-primary"
                                                value="Upload"
                                />
                </fieldset>
</form>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "action": "#",
                "children": <fieldset>
                  <h4>
                                    Upload Payroll Report CSV
                  </h4>
                  <input
                                    id="reportFile"
                                    type="file"
                  />
                  <input
                                    className="button-primary"
                                    type="submit"
                                    value="Upload"
                  />
</fieldset>,
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <fieldset>
                    <h4>
                                        Upload Payroll Report CSV
                    </h4>
                    <input
                                        type="file"
                                        id="reportFile"
                    />
                    <input
                                        type="submit"
                                        class="button-primary"
                                        value="Upload"
                    />
</fieldset>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4>
                        Upload Payroll Report CSV
</h4>,
                      <input
                        id="reportFile"
                        type="file"
/>,
                      <input
                        className="button-primary"
                        type="submit"
                        value="Upload"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <h4>
                        Upload Payroll Report CSV
</h4>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Upload Payroll Report CSV",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Upload Payroll Report CSV",
                      ],
                      "type": "h4",
                    },
                    Object {
                      "instance": <input
                        type="file"
                        id="reportFile"
/>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "reportFile",
                        "type": "file",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "input",
                    },
                    Object {
                      "instance": <input
                        type="submit"
                        class="button-primary"
                        value="Upload"
/>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "button-primary",
                        "type": "submit",
                        "value": "Upload",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "input",
                    },
                  ],
                  "type": "fieldset",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="row"
>
          <div
                    class="column"
          >
                    <form
                              action="#"
                              style="text-align: center;"
                    >
                              <fieldset>
                                        <h4>
                                                  Upload Payroll Report CSV
                                        </h4>
                                        <input
                                                  type="file"
                                                  id="reportFile"
                                        />
                                        <input
                                                  type="submit"
                                                  class="button-primary"
                                                  value="Upload"
                                        />
                              </fieldset>
                    </form>
          </div>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="column"
>
            <form
                        action="#"
                        style={
                                    Object {
                                                "textAlign": "center",
                                              }
                        }
            >
                        <fieldset>
                                    <h4>
                                                Upload Payroll Report CSV
                                    </h4>
                                    <input
                                                id="reportFile"
                                                type="file"
                                    />
                                    <input
                                                className="button-primary"
                                                type="submit"
                                                value="Upload"
                                    />
                        </fieldset>
            </form>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="column"
>
              <form
                            action="#"
                            style="text-align: center;"
              >
                            <fieldset>
                                          <h4>
                                                        Upload Payroll Report CSV
                                          </h4>
                                          <input
                                                        type="file"
                                                        id="reportFile"
                                          />
                                          <input
                                                        type="submit"
                                                        class="button-primary"
                                                        value="Upload"
                                          />
                            </fieldset>
              </form>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                action="#"
                style={
                                Object {
                                                "textAlign": "center",
                                              }
                }
>
                <fieldset>
                                <h4>
                                                Upload Payroll Report CSV
                                </h4>
                                <input
                                                id="reportFile"
                                                type="file"
                                />
                                <input
                                                className="button-primary"
                                                type="submit"
                                                value="Upload"
                                />
                </fieldset>
</form>,
              "className": "column",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <form
                  action="#"
                  style="text-align: center;"
>
                  <fieldset>
                                    <h4>
                                                      Upload Payroll Report CSV
                                    </h4>
                                    <input
                                                      type="file"
                                                      id="reportFile"
                                    />
                                    <input
                                                      type="submit"
                                                      class="button-primary"
                                                      value="Upload"
                                    />
                  </fieldset>
</form>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "action": "#",
                  "children": <fieldset>
                    <h4>
                                        Upload Payroll Report CSV
                    </h4>
                    <input
                                        id="reportFile"
                                        type="file"
                    />
                    <input
                                        className="button-primary"
                                        type="submit"
                                        value="Upload"
                    />
</fieldset>,
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <fieldset>
                      <h4>
                                            Upload Payroll Report CSV
                      </h4>
                      <input
                                            type="file"
                                            id="reportFile"
                      />
                      <input
                                            type="submit"
                                            class="button-primary"
                                            value="Upload"
                      />
</fieldset>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h4>
                          Upload Payroll Report CSV
</h4>,
                        <input
                          id="reportFile"
                          type="file"
/>,
                        <input
                          className="button-primary"
                          type="submit"
                          value="Upload"
/>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": <h4>
                          Upload Payroll Report CSV
</h4>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Upload Payroll Report CSV",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Upload Payroll Report CSV",
                        ],
                        "type": "h4",
                      },
                      Object {
                        "instance": <input
                          type="file"
                          id="reportFile"
/>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "id": "reportFile",
                          "type": "file",
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "input",
                      },
                      Object {
                        "instance": <input
                          type="submit"
                          class="button-primary"
                          value="Upload"
/>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "button-primary",
                          "type": "submit",
                          "value": "Upload",
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "input",
                      },
                    ],
                    "type": "fieldset",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
